plugin/core/windows.py

```diff
@@ -464,6 +464,14 @@ def _apply_workspace_edit(self, params: 'Dict[str, Any]', client: Client, reques
         # reconstruct/get the actual Client object back. Maybe we can (ab)use our homebrew event system for this?
         client.send_response(Response(request_id, {"applied": True}))
 
+    def _get_session_config(self, params: 'Dict[str, Any]', session: Session, client: Client, request_id: int) -> None:
+        items = []  # type: List[Any]
+        requested_items = params.get("items") or []
+        for requested_item in requested_items:
+            items.append(session.config.settings)
```

Comment:
If we will go with this hardcoded solution, maybe we should consider not using root `settings` for that but have a more specific property in `settings` for this case. Should be better in case there is a need later to provide some other settings also.

Reply:
I found no explanation that this configuration request expects a different data structure than the one sent by didChangeConfiguration. With your example, users would suddenly have to move their config into a new property just because the server “upgraded” to the workspace/configuration request.

Reply:
OK, I thought it's some arbitrary settings. Didn't know it's something more specific that is already sent with specific message. Then it makes sense to do it like that.